tsdb/db.go

```diff
@@ -1541,12 +1541,15 @@ func (db *DB) reloadBlocks() (err error) {
 	oldBlocks := db.blocks
 	db.blocks = toLoad
 
-	blockMetas := make([]BlockMeta, 0, len(toLoad))
-	for _, b := range toLoad {
-		blockMetas = append(blockMetas, b.Meta())
-	}
-	if overlaps := OverlappingBlocks(blockMetas); len(overlaps) > 0 {
-		level.Warn(db.logger).Log("msg", "Overlapping blocks found during reloadBlocks", "detail", overlaps.String())
+	// Only check overlapping blocks when overlapping compaction is enabled.
```

Comment:
Why? It seems like a good idea to have this warning especially when overlapping compactions are disabled.

Reply:
The overlapping compaction here means [this](https://github.com/prometheus/prometheus/pull/13393) configuration, not vertical compaction.

I think when overlapping compaction is disabled, it is intentional to have blocks overlapped and TSDB doesn't need to compact it. It is a normal/expected to me so doesn't need to log it as warning or anything.